packages/extension-api/src/extension-api.d.ts

```diff
@@ -3019,6 +3019,17 @@ declare module '@podman-desktop/api' {
      */
     export function getImageInspect(engineId: string, id: string): Promise<ImageInspectInfo>;
 
+    /**
+     * Return low-level information about an image using a {@link ContainerProviderConnection}
+     *
+     * @param containerProviderConnection the connection to the local engine
+     * @param imageName the image name
+     */
+    export function getImageInspect(
+      containerProviderConnection: ContainerProviderConnection,
+      imageName: string,
+    ): Promise<ImageInspectInfo>;
```

Comment:
I think it should return undefined if the image does not exist

Error is if we're not able to grab image or something is invalid but else IMHO it should return undefined if there is no match

Reply:
for getImageInspect(engineId, imageId) usually it should always return a valid image, it's only if you give an invalid imageID that you have an error so you're making an error from the client side

but for the parameter being the image name I would say it's expected to have undefined if there is no match
error would in case we're not able to perform a matching operation

Reply:
I would not see much difference to consider the image name as an identifier at the level than the imageId. 

IMO, when dealing with promises, I always prefer having a catch block, than an if != undefined when dealing with such methods. 

I would be willing to change if other people are in favor (or simply no one care, and you really have this preference) of having undefined added to the return signature.

Reply:
on my side it's a requirement to have `Promise<ImageInspectInfo | undefined>`

Reply:
> on my side it's a requirement to have `Promise<ImageInspectInfo | undefined>`

Ok. I cannot set the return type of the same method name to something different (`Promise<ImageInspectInfo>` and `Promise<ImageInspectInfo | undefined>`, I am getting typescript errors.

To solve this issue, I updated and make a method `findImageInspect`, at the end, I think this is more appropriate.